booking/src/main/java/org/dcsa/conformance/standards/booking/checks/BookingChecks.java

```diff
@@ -0,0 +1,659 @@
+    "Validate ISO Equipment code",
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.ISOEquipmentCode");
+    },
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.ISO_6346_CONTAINER_CODES)
+  );
+
+  private static final JsonContentCheck OUTER_PACKAGING_CODE_IS_VALID = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Validate that 'packagingCode' is a known code",
+    (mav) -> mav.submitAllMatching("requestedEquipments.*.commodities.*.outerPackaging.packageCode"),
+    JsonAttribute.matchedMustBeDatasetKeywordIfPresent(BookingDataSets.OUTER_PACKAGING_CODE)
+  );
+
+  private static Consumer<MultiAttributeValidator> allDg(Consumer<MultiAttributeValidator.AttributePathBuilder> consumer) {
+    return (mav) -> consumer.accept(mav.path("requestedEquipments").all().path("commodities").all().path("outerPackaging").path("dangerousGoods").all());
+  }
+  private static final JsonContentCheck IS_EXPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Export declaration reference ",
+    JsonAttribute.isTrue(JsonPointer.compile("/isExportDeclarationRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/exportDeclarationReference"))
+  );
+
+  private static final JsonContentCheck IS_IMPORT_DECLARATION_REFERENCE_PRESENT = JsonAttribute.ifThen(
+    "Check Import declaration reference presence",
+    JsonAttribute.isTrue(JsonPointer.compile("/isImportLicenseRequired")),
+    JsonAttribute.mustBePresent(JsonPointer.compile("/importLicenseReference"))
+  );
+
+  private static final JsonContentCheck DOCUMENT_PARTY_FUNCTIONS_MUST_BE_UNIQUE = JsonAttribute.customValidator(
+    "Each document party can be used at most once",
+    JsonAttribute.path(
+      "documentParties",
+      JsonAttribute.unique("partyFunction")
+    ));
+
+  private static final JsonContentCheck COMMODITIES_SUBREFERENCE_UNIQUE = JsonAttribute.allIndividualMatchesMustBeValid(
+    "Each Subreference in commodities must be unique",
+    (mav) -> {
+      mav.submitAllMatching("requestedEquipments.*.commodities");
+    },
+    JsonAttribute.unique("subReference")
+  );
+
+
+  private static final JsonContentCheck VALIDATE_SHIPMENT_CUTOFF_TIME_CODE = JsonAttribute.customValidator(
+    "validate shipment cutOff Date time code",
+    (body) -> {
+      var shipmentCutOffTimes = body.path("shipmentCutOffTimes");
+      var receiptTypeAtOrigin = body.path("receiptTypeAtOrigin").asText("");
+      var issues = new LinkedHashSet<String>();
+      var cutOffDateTimeCodes = StreamSupport.stream(shipmentCutOffTimes.spliterator(), false)
+        .map(p -> p.path("cutOffDateTimeCode"))
+        .filter(JsonNode::isTextual)
+        .map(n -> n.asText(""))
+        .collect(Collectors.toSet());
+      if (receiptTypeAtOrigin.equals("CFS")) {
+        if (!cutOffDateTimeCodes.contains("LCL")) {
+          issues.add("For ReceiptTypeAtOrigin as CFS, the shipment cut off date time must have only LCL");
```

Comment:
Message says `only` but the requirement does not prevent other `shipment cut off date time` codes.

Reply:
shipment Cut-off Times / cut-off time code: 	LCL - only applicable when the Receipt Type at Origin is CFS.

Reply:
Again, as I read the code and the issue text, they do not agree.

Reply:
"cutOffDateTimeCode 'LCL' is not present when receiptTypeAtOrigin is CFS"